packages/compiler-core/src/utils.ts

```diff
@@ -56,11 +56,14 @@ export const isBuiltInType = (tag: string, expected: string): boolean =>
 export function isCoreComponent(tag: string): symbol | void {
   if (isBuiltInType(tag, 'Teleport')) {
     return TELEPORT
-  } else if (isBuiltInType(tag, 'Suspense')) {
+  }
+  if (isBuiltInType(tag, 'Suspense')) {
     return SUSPENSE
-  } else if (isBuiltInType(tag, 'KeepAlive')) {
+  } 
+  if (isBuiltInType(tag, 'KeepAlive')) {
     return KEEP_ALIVE
-  } else if (isBuiltInType(tag, 'BaseTransition')) {
+  } 
+  if (isBuiltInType(tag, 'BaseTransition')) {
     return BASE_TRANSITION
```

Comment:
I think it's unnecessary.

Reply:
> I think it's unnecessary.

Yes, i just refactor the codes. It's still working now but code base is not cleanly because i think it has else if hell here. If we use if and return in if block, we can ignore all codes after return and use else if is unnecessarily.

Reply:
FYI https://github.com/vuejs/vue/pull/12874#issuecomment-1320863504